src/helpers/logging-interceptor.helper.ts

```diff
@@ -0,0 +1,19 @@
+import { CallHandler, ExecutionContext, Injectable, Logger, NestInterceptor } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+@Injectable()
+export class LoggingInterceptor implements NestInterceptor {
+  private readonly logger = new Logger();
+
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    const requestBody = context.switchToHttp().getRequest().body;
+    this.logger.debug(`Request body: ${JSON.stringify(requestBody)}`);
+
+    return next.handle().pipe(
+      tap((responseBody) => {
+        this.logger.debug(`Response body: ${JSON.stringify(responseBody)}`);
+      }),
+    );
```

Comment:
This doesn't seem to log the response body for error responses - do we know why?

E.g. this doesn't get triggered on `GET /deals/999` which returns a 404

Reply:
As discussed on call, it's because the `LoggingInterceptor` logs whatever is returned by the controller as the response body, which gets bypassed in an error case (and the response body is instead generated by Nest based on the Exception thrown). As Nest's default logging gives us the status code and we can also see the Exception thrown, we decided that being able to log error bodies gave us no new information, so could be left out.